compiler/modules/front.syntax/parse.bal

```diff
@@ -7,20 +7,49 @@ public type FilePath record {|
     string? directory = ();
 |};
 
-public function parseModulePart(string[] lines, FilePath path, int partIndex) returns ModulePart|err:Syntax {
+public function preParse(string[] lines, FilePath path, int partIndex) returns PreModulePart|err:Syntax {
     SourceFile file = createSourceFile(path, lines);
     Tokenizer tok = new (file);
     check tok.advance();
-    ModulePart part = {
-        file,
-        partIndex,
-        defns: [],
-        importDecl: check parseImportDecl(tok)
-    };
-    while tok.current() != () {
-        part.defns.push(check parseModuleDecl(tok, part));
+    // xxx inline after #443, and check if cloneReadOnly can be avoided
+    ImportDecl? im = check parseImportDecl(tok);
+    return new(file, im == () ? [] : [im.cloneReadOnly()], tok.save(), partIndex);
+}
+
+// xxx alternative names ModulePartPrecursor ModulePrecursor ResolvableModulePart ScannedModulePart
```

Comment:
I think ScannedModulePart is the best. Then you can call the function scanModulePart.